test/coverage/datamodel/PayloadBuilderTest.cpp

```diff
@@ -23,16 +26,667 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	PayloadTest();
+	~PayloadTest() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
+
+	const std::string testStringPayload_;
+	const std::vector<uint8_t> testBytesPayload_ = {'T', 'e', 's', 't'};
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+PayloadTest::PayloadTest()
+    : testStringPayload_("Test"), testBytesPayload_{'T', 'e', 's', 't'} {}
+
+/***********************Serialized Protobuf Tests*****************************/
+
+// Create serialized protobuf payload and verify build payload
+TEST_F(PayloadTest, CreateSerializedProtobufPayloadAndBuildTest) {
+	// Arrange
+	uprotocol::v1::UUri uriObject;
+	uriObject.set_authority_name(testStringPayload_);
+
+	// Act
+	Payload payload(uriObject);
+
+	// Assert
+	auto [payloadData, payloadFormat] = payload.build();
+	EXPECT_EQ(payloadFormat,
+	          uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF);
+	EXPECT_EQ(payloadData, uriObject.SerializeAsString());
+}
+
+// Create serialized protobuf payload with empty message and verify exception
+TEST_F(PayloadTest, CreateEmptySerializedProtobufPayloadExceptionTest) {
+	// Arrange
+	uprotocol::v1::UUri uriObject;
+	uriObject.set_authority_name("");
+
+	// Act
+	Payload payload(uriObject);
+
+	// Assert
+	EXPECT_THROW(payload.build(), Payload::PayloadMoved);
+}
+
+// Create serialized protobuf payload and verify moved payload
+TEST_F(PayloadTest, CreateSerializedProtobufPayloadAndMoveTest) {
+	// Arrange
+	uprotocol::v1::UUri uriObject;
+	uriObject.set_authority_name(testStringPayload_);
+
+	// Act
+	Payload payload(uriObject);
+
+	// Assert
+	auto [payloadData, payloadFormat] = std::move(payload).move();
+
+	EXPECT_EQ(payloadFormat,
+	          uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF);
+	EXPECT_EQ(payloadData, uriObject.SerializeAsString());
```

Comment:
I tried this after reading your suggestion but the test case is failing. I am now looking into this

Reply:
We discussed earlier. The problems seems to be that the test payload string is too short - it's under the _small string optimization_ (SSO) threshold. Depending on platform, compiler, and library version, this is typically 16-32 bytes. Anything under that threshold will be stored in the string object itself instead of dynamically allocated memory, and therefore there will be nothing to move (i.e. the `.data()` address will change).

So long as the string is long enough to be over that threshold, the move tests will work. There are two ways to do this: first is to just use a longer string and second is to use `.reserve()` on the string instance to force it to use a large buffer for a small string.

Reply:
Done